grails-app/jobs/org/pih/warehouse/jobs/AssignIdentifierJob.groovy

```diff
@@ -6,6 +6,11 @@ class AssignIdentifierJob {
 
     static concurrent = false
 
+    // By default this is true on QuartzDisplayJob, which invokes execute()
+    // and if sessionRequired is true, then QuartzDisplayJob tries to do session flush
+    // even if there is no session
+    static sessionRequired = false
```

Comment:
As I wrote in the ticket, the data is persisted without the session. And for the majority of jobs, we don't have to have the session. But in case we might require it then we can do:
```groovy
void execute() {
  Domain.withNewSession {
    ...
  }
}
```
Currently, I am trying to do a `SessionAwareJob`, so we don't have to always remember to do this wrapping if we'd need it.